build/sbt

```diff
@@ -53,7 +53,7 @@ realpath () {
 declare -r noshare_opts="-Dsbt.global.base=project/.sbtboot -Dsbt.boot.directory=project/.boot -Dsbt.ivy.home=project/.ivy"
 declare -r sbt_opts_file=".sbtopts"
 declare -r etc_sbt_opts_file="/etc/sbt/sbtopts"
-declare -r default_sbt_opts="-Xss4m"
+declare -r default_sbt_opts="-Xss256m"
```

Comment:
Can you increase https://github.com/apache/spark/pull/32502/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R2603 too? I have seen Maven build with stacktrace errors. I think it's better to match anyway.

Reply:
BTW, 256m is pretty big. What about 8 or 16? GA machines don't have much memories.

Reply:
Sure, thanks

Reply:
@HyukjinKwon The container for Github action jobs has 7G memory so this should be fine.
The work of https://issues.apache.org/jira/browse/SPARK-35042 is great, but it requires large stack for compiling. There will be more optmizations. Let's make it 256M since the container has enough memory.

Reply:
Also, I already made it 256M for java 11:
https://github.com/apache/spark/pull/32521

Reply:
On second thought, let's reduce it to 16MB since every thread will allocate for such memory for the stack.